### PR TITLE
test: Move My Expenses to Platform: Fixed All Tests

### PR DESCRIPTION
**Pull request by jayfyle**
### <samp>🤖[[deprecated]](https://githubnext.com/copilot-for-prs-sunset) Generated by Copilot at cb8d579</samp>

This pull request enhances the testing and functionality of the `my-expenses.page` module, by adding and modifying test cases for offline expenses, fixing bugs and typos in the page class, and adding mock data for expenses without tx_id.

### <samp>🤖[[deprecated]](https://githubnext.com/copilot-for-prs-sunset) Generated by Copilot at cb8d579</samp>

> _Sing, O Muse, of the valiant coder who dared to test the deletion of expenses_
> _With mock data array, devoid of tx_id, he crafted his cunning assertions_
> _In my-expenses.page.spec.ts he toiled, adding and modifying many a case_
> _Like Hephaestus, the skillful smith, he fixed the bugs and typos with grace_

### <samp>🤖[[deprecated]](https://githubnext.com/copilot-for-prs-sunset) Generated by Copilot at cb8d579</samp>

*  Simplify the access to the platform query params in the `getMyExpenses` function ([link](https://github.com/fylein/fyle-mobile-app/pull/2601/files?diff=unified&w=0#diff-8ba341e210d765eda548d0c9e9ed2568f06e061fafd19b0fc3280e1db80c1533L311-R318))
*  Fix a bug or a typo in the `setExpenseStatsOnSelect` function by replacing the `selectedElements` property with the `selectOutboxExpense` property ([link](https://github.com/fylein/fyle-mobile-app/pull/2601/files?diff=unified&w=0#diff-8ba341e210d765eda548d0c9e9ed2568f06e061fafd19b0fc3280e1db80c1533L934-R934), [link](https://github.com/fylein/fyle-mobile-app/pull/2601/files?diff=unified&w=0#diff-8ba341e210d765eda548d0c9e9ed2568f06e061fafd19b0fc3280e1db80c1533L943-R943))
*  Add a wildcard character to the search string in the `loadMoreExpenses` function to match the `getMyExpenses` function ([link](https://github.com/fylein/fyle-mobile-app/pull/2601/files?diff=unified&w=0#diff-8ba341e210d765eda548d0c9e9ed2568f06e061fafd19b0fc3280e1db80c1533L1444-R1445))
*  Add a new mock data array of expenses without tx_id in `expense.data.ts` for testing the deletion of offline expenses ([link](https://github.com/fylein/fyle-mobile-app/pull/2601/files?diff=unified&w=0#diff-56d972f354883e74d1aa98489079c2d3d173a8d3417fb7d5760cff7a26e7f992R5816-R5820))
*  Add and modify test cases for the `ionViewWillEnter`, `openCreateReportWithSelectedIds`, `deleteSelectedExpenses`, `openDeleteExpensesPopover`, `onSelectAll`, `setOutboxExpenseStatsOnSelect`, and `selectOutboxExpense` functions in `my-expenses.page.spec.ts` to increase the test coverage and handle various scenarios of filtering, selecting, and deleting expenses, especially offline expenses ([link](https://github.com/fylein/fyle-mobile-app/pull/2601/files?diff=unified&w=0#diff-446039c75c9b2d02f68c2a029282002588ffa48a5d20add67bfbce3c26dacbb7R35), [link](https://github.com/fylein/fyle-mobile-app/pull/2601/files?diff=unified&w=0#diff-446039c75c9b2d02f68c2a029282002588ffa48a5d20add67bfbce3c26dacbb7L124-R125), [link](https://github.com/fylein/fyle-mobile-app/pull/2601/files?diff=unified&w=0#diff-446039c75c9b2d02f68c2a029282002588ffa48a5d20add67bfbce3c26dacbb7R936-R953), [link](https://github.com/fylein/fyle-mobile-app/pull/2601/files?diff=unified&w=0#diff-446039c75c9b2d02f68c2a029282002588ffa48a5d20add67bfbce3c26dacbb7R1072), [link](https://github.com/fylein/fyle-mobile-app/pull/2601/files?diff=unified&w=0#diff-446039c75c9b2d02f68c2a029282002588ffa48a5d20add67bfbce3c26dacbb7R1080), [link](https://github.com/fylein/fyle-mobile-app/pull/2601/files?diff=unified&w=0#diff-446039c75c9b2d02f68c2a029282002588ffa48a5d20add67bfbce3c26dacbb7R1125), [link](https://github.com/fylein/fyle-mobile-app/pull/2601/files?diff=unified&w=0#diff-446039c75c9b2d02f68c2a029282002588ffa48a5d20add67bfbce3c26dacbb7R2176-R2186), [link](https://github.com/fylein/fyle-mobile-app/pull/2601/files?diff=unified&w=0#diff-446039c75c9b2d02f68c2a029282002588ffa48a5d20add67bfbce3c26dacbb7L2527-R2573), [link](https://github.com/fylein/fyle-mobile-app/pull/2601/files?diff=unified&w=0#diff-446039c75c9b2d02f68c2a029282002588ffa48a5d20add67bfbce3c26dacbb7R2639-R2661), [link](https://github.com/fylein/fyle-mobile-app/pull/2601/files?diff=unified&w=0#diff-446039c75c9b2d02f68c2a029282002588ffa48a5d20add67bfbce3c26dacbb7L2679-R2744), [link](https://github.com/fylein/fyle-mobile-app/pull/2601/files?diff=unified&w=0#diff-446039c75c9b2d02f68c2a029282002588ffa48a5d20add67bfbce3c26dacbb7R2757-R2768), [link](https://github.com/fylein/fyle-mobile-app/pull/2601/files?diff=unified&w=0#diff-446039c75c9b2d02f68c2a029282002588ffa48a5d20add67bfbce3c26dacbb7L2699-R2776), [link](https://github.com/fylein/fyle-mobile-app/pull/2601/files?diff=unified&w=0#diff-446039c75c9b2d02f68c2a029282002588ffa48a5d20add67bfbce3c26dacbb7R2929-R3029))
*  Remove an empty line in `my-expenses.page.ts` to improve the formatting or style of the code ([link](https://github.com/fylein/fyle-mobile-app/pull/2601/files?diff=unified&w=0#diff-8ba341e210d765eda548d0c9e9ed2568f06e061fafd19b0fc3280e1db80c1533R1390))

## Clickup
https://app.clickup.com/t/86ctv3kk4

## Code Coverage
Please add code coverage here

## UI Preview
Please add screenshots for UI changes

### CODE REVIEW

**Review by OmkarJ13 — CHANGES_REQUESTED** [review]
One comment

File: src/app/fyle/my-expenses/my-expenses.page.spec.ts [hunk 1]
```diff
@@ -2849,4 +2926,105 @@ fdescribe('MyExpensesPage', () => {
...
+
+      expect(component.selectedOutboxExpenses).toEqual([]);
+      expect(component.isReportableExpensesSelected).toBeFalse();
+      expect(component.selectAll).toBeTrue();
+      expect(component.setExpenseStatsOnSelect).toHaveBeenCalledTimes(1);
+      expect(transactionService.isMergeAllowed).toHaveBeenCalledOnceWith([]);
+      expect(component.isMergeAllowed).toBeTrue();
+    });
+
+    it('should remove an expense from selectedOutboxExpenses if it is present in selectedOutboxExpenses', () => {
```
[the remark this answers is not in the document]
Comment (jayfyle) [reply src/app/fyle/my-expenses/my-expenses.page.spec.ts]: There seems to be some issue with the test, checking this separately

File: src/app/fyle/my-expenses/my-expenses.page.spec.ts [hunk 2]
```diff
@@ -2849,4 +2926,105 @@ fdescribe('MyExpensesPage', () => {
...
+      component.selectOutboxExpense(expenseData2);
+
+      const expectedSelectedElements = [...apiExpenseRes, expenseData2];
+      expect(component.selectedOutboxExpenses).toEqual(expectedSelectedElements);
+      expect(component.outboxExpensesToBeDeleted).toEqual(apiExpenseRes);
+      expect(component.cccExpenses).toBe(1);
+      expect(component.selectAll).toBeTrue();
+    });
+
+    it('should remove an expense from selectedOutboxExpenses if it is present in selectedOutboxExpenses and tx_id is not present in expense', () => {
```
Comment (Dimple16) [inline src/app/fyle/my-expenses/my-expenses.page.spec.ts:3012]: are we still using `tx_id`? in platform, shouldn't it just be `id`?

File: src/app/fyle/my-expenses/my-expenses.page.ts [hunk 3]
```diff
@@ -1387,10 +1387,11 @@ export class MyExpensesPage implements OnInit {
       this.trackingService.myExpensesBulkDeleteExpenses({
         count: this.selectedElements.length,
       });
+
       if (data.status === 'success') {
         let totalNoOfSelectedExpenses = 0;
-        if (offlineExpenses?.length > 0) {
-          totalNoOfSelectedExpenses = offlineExpenses?.length + this.selectedElements.length;
+        if (offlineExpenses && offlineExpenses.length > 0) {
```
Comment (Dimple16) [inline src/app/fyle/my-expenses/my-expenses.page.ts:1393]: why change this condition?

**Review by Dimple16 — APPROVED** [review]
added questions, please resolve them before merging

File: src/app/fyle/my-expenses/my-expenses.page.spec.ts (hunk 2 above)
Comment (jayfyle) [reply src/app/fyle/my-expenses/my-expenses.page.spec.ts:3012]: This is for outbox expenses which is still unchanged

**Review by OmkarJ13 — APPROVED** [review]

**Pull request merged**